File: api-escola-conectada/src/main/java/br/edu/escolaconectada/model/Aluno.java
package br.edu.escolaconectada.model;
import jakarta.persistence.*;
@Entity
public class Aluno {
 @Id @GeneratedValue(strategy = GenerationType.IDENTITY) private Long id;
 private String nome; private Integer idade; private String serie;
 @ManyToOne private Responsavel responsavel;
 public Long getId(){return id;} public void setId(Long id){this.id=id;}
 public String getNome(){return nome;} public void setNome(String nome){this.nome=nome;}
 public Integer getIdade(){return idade;} public void setIdade(Integer idade){this.idade=idade;}
 public String getSerie(){return serie;} public void setSerie(String serie){this.serie=serie;}
 public Responsavel getResponsavel(){return responsavel;} public void setResponsavel(Responsavel responsavel){this.responsavel=responsavel;}
}

File: app_escola_conectadaon/lib/models/aluno.dart
import 'responsavel.dart';

class Aluno {
  int? id;
  String nome;
  int idade;
  String serie;
  Responsavel? responsavel;
  Aluno(
      {this.id,
      required this.nome,
      required this.idade,
      required this.serie,
      this.responsavel});
  Map<String, dynamic> toJson() => {
        'id': id,
        'nome': nome,
        'idade': idade,
        'serie': serie,
        'responsavel': responsavel?.toJson()
      };
  factory Aluno.fromJson(Map<String, dynamic> json) => Aluno(
      id: json['id'],
      nome: json['nome'] ?? '',
      idade: json['idade'] ?? 0,
      serie: json['serie'] ?? '',
      responsavel: json['responsavel'] != null
          ? Responsavel.fromJson(json['responsavel'])
          : null);
}
